### DISCUSSION TITLE
Pagination/Count sqls with `union` are really slow

### DISCUSSION THREAD

**Discussion by parallels999** [laravel/framework#46041] [Q&A]
Current SQL looks like this
```sql
select count(*) as aggregate from (
    (select * from `audits_1`) 
    union (select * from `audits_2`)
) as `temp_table`; /* 64.21ms */
```
How about querying each sql individually by union for avoid getting all the columns before the count?
```sql
select count(*) as aggregate from `audits_1`; /* 5.29ms */
select count(*) as aggregate from `audits_2`; /* 5.76ms */
```
Sow now we can accumulate results (count_0 + count_1 + count_n)

Also
```sql
select (
     select count(*) as aggregate from `audits_1`
     + select count(*) as aggregate from `audits_2`
) as aggregate; /* 6.16ms */
```

https://github.com/laravel/framework/blob/df696e9963903172ea218bc97d00b72ce738c0ee/src/Illuminate/Database/Query/Builder.php#L2789-L2794

**Comment by ImBIOS** [comment]
```php
$audits_1_count = DB::table('audits_1')->count();
$audits_2_count = DB::table('audits_2')->count();

$total_count = $audits_1_count + $audits_2_count;

$result = DB::table(DB::raw(''))
             ->selectRaw('(?) as aggregate', [$total_count])
             ->get();
```

@parallels999 Would not the above Laravel code is similar to:

```sql
select (
     select count(*) as aggregate from `audits_1`
     + select count(*) as aggregate from `audits_2`
) as aggregate;
```

I wonder if it does faster tho. How can I measure the processing time of the above code?

**Reply by parallels999** [reply]
I'm using laravel debugbar, it show times of every sql

**Reply by ImBIOS** [reply]
@parallels999 Can you share your project specific to this issue so we'll quickly reproduce it and determine the best solution?
It'd be awesome if we could make Laravel even faster and more reliable. I love 💝 your finding.
